core/templates/services/user.service.ts

```diff
@@ -56,21 +60,32 @@ export class UserService {
     return this.userInfo;
   }
 
-  async getProfileImageDataUrlAsync(): Promise<string> {
-    let defaultUrl = (
-      this.urlInterpolationService.getStaticImageUrl(
-        AppConstants.DEFAULT_PROFILE_IMAGE_PATH));
-    return this.getUserInfoAsync().then(
-      async(userInfo) => {
-        if (userInfo.isLoggedIn()) {
-          return this.userBackendApiService.getProfileImageDataUrlAsync(
-            defaultUrl);
-        } else {
-          return new Promise((resolve, reject) => {
-            resolve(defaultUrl);
-          });
-        }
-      });
+  getProfileImageDataUrlAsync(username: string): [string, string] {
+    let defaultUrlWebp = this.urlInterpolationService.getStaticImageUrl(
+      AppConstants.DEFAULT_PROFILE_IMAGE_WEBP_PATH);
+    let defaultUrlPng = this.urlInterpolationService.getStaticImageUrl(
+      AppConstants.DEFAULT_PROFILE_IMAGE_PNG_PATH);
+    // We store profile images to the browser-local-storage when in the
+    // emulator mode. In general we return the tuple of png image URL
+    // and web image url but as we are only storing the image in the
+    // local storage we return tuple of localStoredImage, one for png
+    // and other for webp.
+    let localStoredImage = this.imageLocalStorageService.getRawImageData(
+      username + '_profile_picture.png');
+    if (localStoredImage !== null) {
+      return [localStoredImage, localStoredImage];
+    }
+    if (AppConstants.EMULATOR_MODE) {
+      return [defaultUrlPng, defaultUrlWebp];
+    } else {
+      let pngImageUrl = this.urlInterpolationService.interpolateUrl(
+        this.assetsBackendApiService.profileImagePngUrlTemplate,
+        {username: username});
+      let WebpImageUrl = this.urlInterpolationService.interpolateUrl(
+        this.assetsBackendApiService.profileImageWebpUrlTemplate,
+        {username: username});
+      return [pngImageUrl, WebpImageUrl];
+    }
```

Comment:
```suggestion
    if (AppConstants.EMULATOR_MODE) {
      let localStoredImage = this.imageLocalStorageService.getRawImageData(
        username + '_profile_picture.png');
      if (localStoredImage == null) {
        return [defaultUrlPng, defaultUrlWebp];
      }
      return [localStoredImage, localStoredImage];
    } else {
      let pngImageUrl = this.urlInterpolationService.interpolateUrl(
        this.assetsBackendApiService.profileImagePngUrlTemplate,
        {username: username});
      let WebpImageUrl = this.urlInterpolationService.interpolateUrl(
        this.assetsBackendApiService.profileImageWebpUrlTemplate,
        {username: username});
      return [pngImageUrl, WebpImageUrl];
    }
```

And move the image above the `return [localStoredImage, localStoredImage];`

Reply:
Done